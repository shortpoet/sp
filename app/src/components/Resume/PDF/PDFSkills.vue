<template>
  <section class="resume-section mt-2 ml-0" id="skills">
    <div class="my-auto">
      <h4 style="font-family: 'Saira Extra Condensed';" class="heading heading-4 subheading mb-4 mb-md-3 ml-4">
        Programming Languages &amp; Tools
      </h4>
      <div id="skill-grid-container" class="d-flex flex-column flex-md-row justify-content-between mt-2 mx-5 px-0">
        <div
          v-for="(type, i) in skills"
          :key="i"
          class="skill-grid"
        >
          <div class="list-devicons devicon-row d-flex justify-content-around my-0">
            <PDFDevIcon
              v-for="(icon, i) in mapIcons(type.type)"
              :key="i"
              :source="icon.icon"
              :name="icon.name"
            />
          </div>
          <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
            <div>
              <div class="pdf-skill-type" style="font-family: 'Open Sans';">{{ type.type }}</div>
            </div>
          </div>
          <PDFBorder class="d-none d-md-block my-3 my-md-2" :size=".25"/>
          <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
              <div
                v-for="(skill, i) in listSkills(type.details)"
                :key="i"
                class="skill-list d-flex flex-column mx-2 mb-2"
              >
                <div class="skill-pill-container d-flex justify-content-around">
                  <!-- adding extra spans to create grid -->
                  <span></span>
                  <span style="font-family: 'Open Sans';" class="skill badge badge-pill">
                    {{ skill }}
                  </span>
                  <span></span>
                </div>          
              </div>
          </div>
          <PDFBorder v-if="i !== (skills.length - 1)" class="d-md-none d-block my-3 my-md-2" :marginX="25" :size=".25"/>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import PDFDevIcon from '@/components/Resume/PDF/PDFDevIcon'
import PDFBorder from '@/components/Resume/PDF/PDFBorder'
import icons from '@/assets/icons.js'

export default {
  name: 'PDFSkills',
  props: {
    skills: {
      type: Array
    }
  },
  components: {
    PDFDevIcon,
    PDFBorder
  },
  data () {
    return {
      icons: icons.icons,
      iconMap: icons.iconMap,
      windowWidth: window.innerWidth 
    }
  },
  computed: {
  },
  methods: {
    listSkills(skills) {
      return skills.split(',')
    },
    mapIcons(skill) {
      console.log(skill)
      let iconKeys = this.iconMap.filter(im => im.skill === skill)[0]['icons']
      return iconKeys.map(ik => {
        return this.icons.filter(i => i.name === ik)[0]
      }) 
    }
  },
  mounted () {
  }
}
</script>
<style lang="scss">
.fa-user-ninja:before {
  font-family: 'Font Awesome 5 Free';
  content: "\f504"; 
  }
</style>