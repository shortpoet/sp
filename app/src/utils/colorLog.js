const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
