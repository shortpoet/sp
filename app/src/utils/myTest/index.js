import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
