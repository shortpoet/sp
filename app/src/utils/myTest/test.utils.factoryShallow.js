import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow