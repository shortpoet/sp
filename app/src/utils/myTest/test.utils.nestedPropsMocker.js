

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
