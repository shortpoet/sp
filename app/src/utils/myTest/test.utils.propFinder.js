import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder