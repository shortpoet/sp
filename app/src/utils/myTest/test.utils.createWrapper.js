import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper