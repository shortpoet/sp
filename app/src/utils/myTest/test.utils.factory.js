import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory