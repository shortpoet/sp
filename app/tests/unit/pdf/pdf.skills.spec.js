import * as myTest from '@/utils/myTest'
const { createWrapper, propsFinder } = myTest.default
import PDFSkills from '@/components/Resume/PDF/PDFSkills'
import { cloneDeep } from 'lodash'
// import icons from '@/assets/icons.js'

describe('PDFSkills.vue', () => {

  const component = PDFSkills

  let wrapper
  const mountOptions = propsFinder(['skills'])
  let expected

  beforeEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
    expected = ``
  })

  describe('element and class checks', () => {

    it('renders .pdf-skill-type div for each type of skill', () => {
      
      expected += `pdf-skill-type`
      wrapper = createWrapper(component, mountOptions)

      expect(wrapper.find('.pdf-skill-type').attributes().class).toContain(expected)
  
    })

    it('matches snapshot', () => {

      wrapper = createWrapper(component, mountOptions)
  
      expect(wrapper.html()).toMatchSnapshot()
  
    })

  })

  
  // icons.forEach(icon => {
  //   // describe('icons')
  // })

})
