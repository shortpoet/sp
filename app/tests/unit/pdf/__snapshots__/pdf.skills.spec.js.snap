// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFSkills.vue element and class checks matches snapshot 1`] = `
<section id="skills" class="resume-section mt-2 ml-0">
  <div class="my-auto">
    <h4 class="heading heading-4 subheading mb-4 mb-md-3 ml-4" style="font-family: 'Saira Extra Condensed';">
      Programming Languages &amp; Tools
    </h4>
    <div id="skill-grid-container" class="d-flex flex-column flex-md-row justify-content-between mt-2 mx-5 px-0">
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-dotnet"></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Dotnet</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  Framework 4.x
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Core
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Entity Framework
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Razor
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   MVC
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-javascript"></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Javascript</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  Vue
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Typescript
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Leafly
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Plotly
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   c3
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-python"></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Python</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  Pandas
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Matplotlib
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Flask
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Selenium
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Sklearn
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-msql_server"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-mysql"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-sqllite"></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Relational Db</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  msSQL
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   mySQL
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Entity Framework
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   SQLAlchemy
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-mongodb"></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Document Db</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  mongoDB
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Pymongo
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-html5"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-css3"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-bootstrap"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-sass"></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Markdown</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  HTML
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   CSS
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   YAML
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   MD
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   SCSS
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class=""></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Data Visualization</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  D3
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Tableau
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-visualstudio"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-git"></i>
            <!---->
          </div>
          <div><i class="fa fa-cogs"></i>
            <!---->
          </div>
          <div><i class="devicons devicons-linux"></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">Workflow</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  Mobile-First &amp; Responsive Design
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Cross Browser Testing &amp; Debugging
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Agile Development &amp; Scrum
                </span> <span></span></div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 6.25rem;" class="d-md-none d-block my-3 my-md-2">
      </div>
      <div class="skill-grid">
        <div class="list-devicons devicon-row d-flex justify-content-around my-0">
          <div><i class="devicons devicons-docker"></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
          <div><i class="devicons devicons-aws"></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
          <div><i class="devicons devicons-go"></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
          <div><i class=""></i>
            <!---->
          </div>
        </div>
        <div class="skill-type d-flex flex-row align-items-center justify-content-around my-2 my-md-0">
          <div>
            <div class="pdf-skill-type" style="font-family: 'Open Sans';">DevOps</div>
          </div>
        </div>
        <hr style="border-width: 0.0625rem; margin: 0.5rem;" class="d-none d-md-block my-3 my-md-2">
        <div class="skill-list-container d-flex flex-column mb-4 mb-md-2">
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                  Docker
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Azure DevOps CI/CD Build &amp; Release Pipelines
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   AWS
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Azure Resource Manager
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Yeoman Templates
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Terraform
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Terratest
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Go
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Powershell
                </span> <span></span></div>
          </div>
          <div class="skill-list d-flex flex-column mx-2 mb-2">
            <div class="skill-pill-container d-flex justify-content-around"><span></span> <span class="skill badge badge-pill" style="font-family: 'Open Sans';">
                   Bash
                </span> <span></span></div>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
</section>
`;
